code/API_definitions/number-recycling.yaml

```diff
@@ -0,0 +1,224 @@
+openapi: 3.0.3
+info:
+  title: Number Recycling
+  description: |
+    The API can be used to check whether the subscriber of the phone number has changed.
+    A common scenario is when a fraudster obtains a phone number that previously belonged to another person who did not properly deactivate or verify it. The fraudster can then use this number to commit identity theft or engage in fraudulent activities, such as obtaining access codes for accounts associated with that number. In some cases, fraudsters may also attempt to use services or avoid charges by leveraging old, no longer used numbers that were previously used by other users.
+    Response includes whether there will be changes from the specified date.
+    
+  version: wip
+  x-camara-commonalities: 0.4.0
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+externalDocs:
+  description: Product documentation at Camara
+  url: https://github.com/camaraproject/NumberRecycling
+servers:
+  - url: '{apiRoot}/number-recycling/v0.1'
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root, defined by the service provider, e.g. `api.example.com` or `api.example.com/somepath`
+paths:
+  /check:
+    post:
+      security:
+        - openId:
+          - number-recycling:check
+      tags:
+        - Check Number Recycling
+      description: Check whether the subscriber of the phone number has changed.
+      operationId: checkNumberRecycling
+      parameters:
+        - $ref: '#/components/parameters/x-correlator'      
+      requestBody:
+        description: |
+          Check whether the subscriber of the phone number has changed.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateCheckNumRecycling'
+        required: true
+      responses:
+        '200':
+          $ref: '#/components/responses/Generic200'
+        '400':
+          $ref: '#/components/responses/Generic400'
+        '401':
+          $ref: '#/components/responses/Generic401'
+        '403':
+          $ref: '#/components/responses/Generic403'
+        '404':
+          $ref: '#/components/responses/Generic404'
+        '500':
+          $ref: '#/components/responses/Generic500'
+        '503':
+          $ref: '#/components/responses/Generic503'
+components:
+  securitySchemes:
+    openId:
+      type: openIdConnect
+      openIdConnectUrl: https://example.com/.well-known/openid-configuration
+  parameters:
+    x-correlator:
+      name: x-correlator
+      in: header
+      description: Correlation id for the different services
+      schema:
+        type: string
+  headers:
+    x-correlator:
+      description: Correlation id for the different services
+      schema:
+        type: string
+  schemas:
+    CreateCheckNumRecycling:
+      type: object
+      required:
+        - phoneNumber
```

Comment:
@eric-murray 
Because we assume that this API uses 2-legged access token.
As the user story mentioned, ASP checks whether there has been a change in the user associated with the phone number after the specified date so that API can ensure that a phone number is correctly linked to an user and prevent mis-delivery before sending SMS messages to an wrong user (if there has been a change).
If we apply 3-legged access token, we cannot use this API. Because ASP may authenticate the wrong user and potentially may ask for the wrong user's consent for the API access with 3-legged access token.

Reply:
Hello @eric-murray 
Thank for your comment.

As the description in SIM Swap API yaml file said, 2-legged access tokens can be applicable as one of the options.
If my understanding is incorrect, please let me know.

 >### Restrictions for tokens without an associated authenticated phone number:
 >   - For scenarios which do not have a phone number associated to the token during the authentication flow, e.g. 2-legged access tokens, the `phoneNumber` field MUST be provided in the API request. This ensures that the phone number is explicit and valid for each API call made with these tokens.

Reply:
Hi @yamamoto0104

Indeed, if the SIM Swap API is called with a 2-legged access token, then `phoneNumber` must be provided. But if it is called with a 3-legged token, then a `phoneNumber` MUST NOT be provided. So the `phoneNumber` is not a `required` parameter. This can be seen from the SIM Swap [`CreateCheckSimSwap`](https://github.com/camaraproject/SimSwap/blob/main/code/API_definitions/sim-swap.yaml#L241#L254) schema.

In your proposed API definition, `phoneNumber` is always a required parameter. So the API consumer would be forced to provide it, irrespective of whether at 2- or 3-legged token was being used.

Reply:
Hi @eric-murray 
Thank you for your kind response!
For Number Recycling API, we might be able to take the same approach as SIM Swap API. We'll discuss it internally and feedback later.

Reply:
Hi @eric-murray 

We updated the phone number description based on the Appendix of API design guidelines. We also added 422 error code based on the Appendix.

[Appendix A: info.description template for device identification from access token](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#appendix-a-infodescription-template-for-device-identification-from-access-token)